src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1046 @@
+        }
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Locking is necessary because statistics is still being collected there and if the device is in the call
+     * and collecting some information available while in the call only - it'll end up with a 404 error.
+     * If there's a conferenceId available - only one conference is removed, otherwise - the method
+     * iterates through all of the available conferences and removes them.
+     */
+    @Override
+    public void hangup(String conferenceId) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Hangup string received: " + conferenceId);
+        }
+        checkAndInitialize();
+        controlOperationsLock.lock();
+        try {
+            if (!StringUtils.isNullOrEmpty(conferenceId)) {
+                doDelete(String.format(CONFERENCES, conferenceId));
+            } else {
+                ArrayNode conferenceCall = listConferenceCalls();
+                for (JsonNode node : conferenceCall) {
+                    Integer id = getJsonProperty(node, "id", Integer.class);
+                    doDelete(String.format(CONFERENCES, id));
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+    }
+
+    @Override
+    public CallStatus retrieveCallStatus(String conferenceId) throws Exception {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Retrieving call status with string: " + conferenceId);
+        }
+        checkAndInitialize();
+        controlOperationsLock.lock();
+        try {
+            if(!StringUtils.isNullOrEmpty(conferenceId)){
+                JsonNode response = doGet(String.format(CONFERENCES, conferenceId), JsonNode.class);
+                if (response == null) {
+                    return generateCallStatus(conferenceId, CallStatus.CallStatusState.Disconnected);
+                }
+                Boolean conferenceIsActive = getJsonProperty(response, "isActive", Boolean.class);
+                if (conferenceIsActive != null && conferenceIsActive) {
+                    return generateCallStatus(getJsonProperty(response, "id", String.class), CallStatus.CallStatusState.Connected);
+                }
+            } else {
+                ArrayNode conferences = listConferenceCalls();
+                if(conferences != null && conferences.size() > 0){
+                    return generateCallStatus(getJsonProperty(conferences.get(0), "id", String.class), CallStatus.CallStatusState.Connected);
+                }
+            }
+        } finally {
+            controlOperationsLock.unlock();
+        }
+        return generateCallStatus(conferenceId, CallStatus.CallStatusState.Disconnected);
+    }
+
+    @Override
+    public MuteStatus retrieveMuteStatus() throws Exception {
+        checkAndInitialize();
+        Boolean muted = doGet(AUDIO_MUTED, Boolean.class);
+        if (muted) {

```

Comment:
Source of NullPointerException during unboxing. Guys, I sent tens of the emails on this topic. NEVER use unboxing, this is the worst feature ever in programming, it caused (and keeps causing) many thousands bugs worldwide. Boxing is OK, unboxing is not.